apps/hash-graph/bin/cli/src/main.rs

```diff
@@ -3,29 +3,51 @@
 
 mod args;
 mod error;
+mod parser;
 mod subcommand;
 
-use error_stack::Result;
+use std::sync::Arc;
+
+use error_stack::{Report, Result};
 use graph::load_env;
 
-use self::{args::Args, error::GraphError, subcommand::Subcommand};
+use self::{args::Args, error::GraphError};
+use crate::error::SentryError;
 
-#[tokio::main]
-async fn main() -> Result<(), GraphError> {
+fn main() -> Result<(), GraphError> {
     load_env(None);
 
-    let args = Args::parse_args();
+    let Args {
+        sentry_dsn,
+        subcommand,
+    } = Args::parse_args();
+
+    if let Some(dsn) = &sentry_dsn {
+        let client = sentry::Client::from_config(dsn);
 
-    match args.subcommand {
-        Subcommand::Server(args) => subcommand::server(args).await,
-        Subcommand::Migrate(args) => subcommand::migrate(args).await,
-        Subcommand::TypeFetcher(args) => subcommand::type_fetcher(args).await,
-        Subcommand::Completions(ref args) => {
-            subcommand::completions(args);
-            Ok(())
+        if !client.is_enabled() {
+            return Err(Report::new(SentryError::InvalidDsn).change_context(GraphError));
```

Comment:
Let's attach the DSN. These kinds of attachments makes life so much easier with `error-stack` and this should be easy enough I think.

Reply:
Good call! ([a5b4bcb](https://github.com/hashintel/hash/pull/3091/commits/a5b4bcbb75caaf586b4c96766d4ce6d094380973))